src/elements/content-sharing/ContentSharing.js

```diff
@@ -0,0 +1,171 @@
+    const { apiHost, itemID, itemType, language, token } = props;
+    const [item, setItem] = useState<?itemFlowType>(null);
+    const [sharedLink, setSharedLink] = useState<?ContentSharingSharedLinkType>(null);
+    const [currentUserID, setCurrentUserID] = useState<?string>(null);
+    const [errorExists, setErrorExists] = useState<boolean>(false);
+
+    const api = new API({
+        apiHost,
+        clientName: CLIENT_NAME_CONTENT_SHARING,
+        id: `${itemType}_${itemID}`,
+        token,
+    });
+
+    const getError = useCallback(() => {
+        setErrorExists(true);
+    }, [setErrorExists]);
+
+    useEffect(() => {
+        setItem(null);
+        setSharedLink(null);
+        setCurrentUserID(null);
+    }, [token, itemID, itemType]);
+
+    useEffect(() => {
+        const getItemSuccess = (itemData: ContentSharingItemAPIResponse) => {
+            const { item: itemFromAPI, sharedLink: sharedLinkFromAPI } = convertItemResponse(itemData);
+            setItem(itemFromAPI);
+            setSharedLink(sharedLinkFromAPI);
+            setErrorExists(false);
+        };
+
+        const getItem = async () => {
+            if (itemType === TYPE_FILE) {
+                await api.getFileAPI().getFile(itemID, getItemSuccess, getError, {
+                    fields: [
+                        FIELD_ALLOWED_INVITEE_ROLES,
+                        FIELD_DESCRIPTION,
+                        FIELD_EXTENSION,
+                        FIELD_ID,
+                        FIELD_NAME,
+                        FIELD_PERMISSIONS,
+                        FIELD_SHARED_LINK,
+                        FIELD_SHARED_LINK_FEATURES,
+                        FIELD_ITEM_TYPE,
+                    ],
+                });
+            }
+
+            if (itemType === TYPE_FOLDER) {
+                await api.getFolderAPI().getFolderFields(itemID, getItemSuccess, getError, {
+                    fields: [
+                        FIELD_ALLOWED_INVITEE_ROLES,
+                        FIELD_DESCRIPTION,
+                        FIELD_EXTENSION,
+                        FIELD_ID,
+                        FIELD_NAME,
+                        FIELD_PERMISSIONS,
+                        FIELD_SHARED_LINK,
+                        FIELD_SHARED_LINK_FEATURES,
+                        FIELD_ITEM_TYPE,
+                    ],
+                });
+            }
+        };
+
+        if (!item && !sharedLink) {
+            getItem();
+        }
+    }, [api, getError, item, itemID, itemType, sharedLink, currentUserID]);
+
+    useEffect(() => {
+        const getUserSuccess = userData => {
+            const { id, userEnterpriseData } = convertUserResponse(userData);
+            setSharedLink({ ...sharedLink, ...userEnterpriseData });
+            setCurrentUserID(id);
+            setErrorExists(false);
+        };
+
+        const getUserData = async () => {
+            await api.getUsersAPI(false).getUser(itemID, getUserSuccess, getError, {
```

Comment:
getUsersAPI.getUser takes in a callback for success and returns a promise?  Ideally it would just return a promise that you can use to get the success/failure state.  This is probably pre-existing functionality though

Reply:
Yes, good catch. I removed the async/await wrappers from the API calls.